api/v1alpha1/promise_types.go

```diff
@@ -244,6 +244,17 @@ func (p *Promise) GetPipelineResourceName() string {
 func (p *Promise) GetPipelineResourceNamespace() string {
 	return "default"
 }
+func (p *Promise) GetDynamicControllerName() string {
+	name := p.GetName()
+	// We only start a dynamic controller if the promise contains an API
+	// so this **should** always be safe
+	if _, crd, err := p.GetAPI(); err == nil {
+		//using _ as a separator because its not a valid character in a k8s name
+		//therefore it should never cause a conflict with a CRD name or a promise name
+		name += "_" + crd.GetName()
+	}
+	return name
+}
```

Comment:
Understand that promises will have a crd here following the current code path, but still find it a bit risky to silently ignore an error that we receive.

May I sugges logging this error? That way we are not completely ignoring it.